example/main.go

```diff
@@ -2,152 +2,262 @@ package main
-	if err != nil {
-		log.Fatalf("Failed to create azure request: %v", err)
+
+	// Load configuration from environment variables
+	bedrockConfig := BedrockConfig{
+		AccessKeyID:     os.Getenv("AWS_ACCESS_KEY_ID"),
+		SecretAccessKey: os.Getenv("AWS_SECRET_ACCESS_KEY"),
+		Region:          os.Getenv("AWS_REGION"),
 	}
-	vertexRequest, err := vertex.NewRequest(cfg.VertexProjectID, cfg.VertexLocation)
-	if err != nil {
-		log.Fatalf("Failed to create vertex request: %v", err)
+
+	// Check if Bedrock credentials are available
+	if bedrockConfig.AccessKeyID == "" || bedrockConfig.SecretAccessKey == "" || bedrockConfig.Region == "" {
+		log.Fatalf("AWS Bedrock credentials not found in environment variables")
 	}
 
-	// Get model configuration
-	modelConfig := config.GetModelConfig()
-	modelConfig.Clients = []http.Request{
-		*openaiRequest,
-		*azureRequest,
-		*vertexRequest,
+	fmt.Println("🔑 AWS Bedrock credentials loaded successfully")
+	fmt.Printf("Using AWS region: %s\n", bedrockConfig.Region)
+
+	// Test Bedrock API with Claude model
+	modelID := "anthropic.claude-3-sonnet-20240229-v1:0"
+
+	// Create request payload for Claude
+	request := ClaudeRequest{
+		AnthropicVersion: "bedrock-2023-05-31",
+		MaxTokens:        1000,
+		Messages: []map[string]interface{}{
+			{
+				"role":    "user",
+				"content": "Hello, how are you? Tell me a short joke.",
+			},
+		},
 	}
-	modelConfig.RedisConfig = &cfg.RedisConfig
 
-	// Set up Azure regions
-	modelConfig.AzureRegions = map[string]string{
-		"eastus":     cfg.AzureEndpoint,
-		"westeurope": "https://custom-westeurope.openai.azure.com",
+	// Make request to Bedrock API
+	response, err := invokeBedrockModel(bedrockConfig, modelID, request)
+	if err != nil {
+		log.Fatalf("Failed to invoke Bedrock model: %v", err)
 	}
 
-	// Create transport with configuration
-	transport, err := transport.NewTransport(modelConfig)
+	// Print response
+	fmt.Println("\n🤖 AWS Bedrock Response:")
+	fmt.Println(response)
+}
+
+// invokeBedrockModel makes a request to the AWS Bedrock API
+func invokeBedrockModel(config BedrockConfig, modelID string, request interface{}) (string, error) {
+	// Convert request to JSON
+	jsonData, err := json.Marshal(request)
 	if err != nil {
-		log.Fatalf("Failed to create transport: %v", err)
+		return "", fmt.Errorf("failed to marshal request: %v", err)
 	}
 
-	// Create HTTP client with our transport
-	client := &http.Client{
-		Transport: transport,
+	// Create request URL and host
+	host := fmt.Sprintf(awsBedrockHost, config.Region)
+	path := fmt.Sprintf(awsBedrockPath, modelID)
+	url := fmt.Sprintf("https://%s%s", host, path)
+
+	// Get current time for signing
+	now := time.Now().UTC()
+	amzDate := now.Format("20060102T150405Z")
+	dateStamp := now.Format("20060102")
+
+	// Calculate payload hash
+	payloadHash := hashSHA256(jsonData)
+
+	// Create HTTP request
+	req, err := http.NewRequest("POST", url, bytes.NewBuffer(jsonData))
```

Comment:
First I'd use `bytes.Reader` instead; buffer is for when you want to be able to modify the underlying byte array, but in this case it's just reading.

```suggestion
	req, err := http.NewRequest("POST", url, bytes.NewReader(jsonData))
```

Second, I'd use the one that supports a context (https://pkg.go.dev/net/http#NewRequestWithContext) so you can set a timeout:

```suggestion
	ctx, cancel := context.WithTimeout(context.Background, 5 * time.Second)
	defer cancel()
	
	req, err := http.NewRequestWithContext(ctx, "POST", url, bytes.NewBuffer(jsonData))
```